src/plugins/mer/mersettings.cpp

```diff
@@ -256,18 +363,83 @@ void MerSettings::read()
             break;
 
         const QVariantMap deviceModelData = data.value(key).toMap();
-        MerEmulatorDeviceModel deviceModel;
+        MerEmulatorDeviceModel deviceModel(isSdkProvided);
         deviceModel.fromMap(deviceModelData);
 
-        m_deviceModels.insert(deviceModel.name(), deviceModel);
+        result.insert(deviceModel.name(), deviceModel);
     }
+
+    return result;
 }
 
-void MerSettings::save()
```

Comment:
Similarly to read(), could the diff be minimized?

Reply:
Reworked. Diff is minimal.